libs/feature/record/src/lib/map-view/map-view.component.html

```diff
@@ -25,14 +25,56 @@
       class="top-[1em] right-[1em] p-3 bg-white absolute overflow-y-auto overflow-x-hidden max-h-72 w-56"
       [class.hidden]="!selection"
     >
-      <button
-        (click)="resetSelection()"
-        class="rounded bg-primary-opacity-25 text-white absolute right-[0.5em]"
+      <gn-ui-button
+        type="light"
+        (buttonClick)="resetSelection()"
+        style="
+          --gn-ui-button-padding: 0px;
+          --gn-ui-button-width: 24px;
+          --gn-ui-button-height: 24px;
+        "
+        extraClass="absolute right-[0.5em] ml-[8px] mr-[10px]"
       >
         <ng-icon name="matClose" class="align-middle text-sm"></ng-icon>
-      </button>
+      </gn-ui-button>
       <gn-ui-feature-detail [feature]="selection"></gn-ui-feature-detail>
     </div>
+
+    <div
+      class="top-[1em] p-3 bg-white absolute overflow-y-auto overflow-x-hidden max-h-72 w-56"
+      [ngClass]="{ 'right-[1em]': !selection, 'right-[16em]': selection }"
+      [hidden]="!showLegend || !legendExists"
+    >
+      <div class="flex justify-between items-center mb-2">
+        <div class="text-primary font-bold">Legend</div>
+        <gn-ui-button
+          type="light"
+          (buttonClick)="toggleLegend()"
+          style="
+            --gn-ui-button-padding: 0px;
+            --gn-ui-button-width: 24px;
+            --gn-ui-button-height: 24px;
+          "
+          extraClass="ml-[8px] mr-[10px]"
+        >
+          <ng-icon name="matClose" class="align-middle text-sm"></ng-icon>
+        </gn-ui-button>
+      </div>
+      <gn-ui-map-legend
+        [context]="mapContext$ | async"
```

Comment:
The test is failing because this adds another subscription to `mapContext$`, which triggers the extent computation one more time; in the tests we only resolve the extent computation promise once, so the other does not complete and as such one of the subscriptions doesn't receive the full computed context.

Long story short, I think a good fix is to add this:
```diff
    withLatestFrom(this.mdViewFacade.metadata$),
    map(([context, metadata]) => {
      if (context.view) return context
      const extent = this.mapUtils.getRecordExtent(metadata)
      const view = extent ? { extent } : null
      return {
        ...context,
        view,
      }
    }),
+   shareReplay(1) // line 172
  )
```

Then we won't do unnecessary extent computations :)

Reply:
Thanksa a lot :)